bounds.py

```diff
@@ -337,7 +395,8 @@ def gram_iteration_on_matrix_explicit_backward(M, n_iter=100, return_time=True):
 
 
 def compute_singla2021(X, n_iter=50, return_time=True, device="cuda"):
-    """Estimate spectral norm of convolutional layer with Singla2021.
+    """Estimate spectral norm of convolutional layer with circular padding 
+    with Singla2021.
 
     From a convolutional filter, this function estimates the spectral norm of
     the convolutional layer using Singla2021 [1]_.
```

Comment:
Short summary is a one-line summary.
Give more details in extended summary.
```suggestion
    """Estimate spectral norm of convolutional layer with Singla2021.

    From a convolutional filter, this function estimates the spectral norm of
    the convolutional layer for circular padding using Singla2021 [1]_.
```